src/Parser/PhpDocParser.php

```diff
@@ -60,8 +60,38 @@ private function parseChild(TokenIterator $tokens): Ast\PhpDoc\PhpDocChildNode
 
 	private function parseText(TokenIterator $tokens): Ast\PhpDoc\PhpDocTextNode
 	{
-		$text = $tokens->joinUntil(Lexer::TOKEN_PHPDOC_EOL, Lexer::TOKEN_CLOSE_PHPDOC, Lexer::TOKEN_END);
-		$text = rtrim($text, " \t"); // the trimmed characters MUST match Lexer::TOKEN_HORIZONTAL_WS
+		$text = '';
+		while (true) {
+			// If we received a Lexer::TOKEN_PHPDOC_EOL, exit early to prevent
+			// them from being processed.
+			if ($tokens->currentTokenType() === Lexer::TOKEN_PHPDOC_EOL) {
+				break;
+			}
+			$text .= $tokens->joinUntil(Lexer::TOKEN_PHPDOC_EOL, Lexer::TOKEN_CLOSE_PHPDOC, Lexer::TOKEN_END);
+			$text = rtrim($text, " \t");
+
+			// If we joined until TOKEN_PHPDOC_EOL, peak at the next tokens to see
+			// if we have a multiline string to join.
+			if ($tokens->currentTokenType() !== Lexer::TOKEN_PHPDOC_EOL) {
+				break;
+			}
+
+			// Peek at the next token to determine if it is more text that needs
+			// to be combined.
+			$tokens->pushSavePoint();
+			$tokens->next();
+			if ($tokens->currentTokenType() === Lexer::TOKEN_PHPDOC_EOL) {
+				$tokens->next();
+			}
+			if ($tokens->currentTokenType() !== Lexer::TOKEN_IDENTIFIER) {
+				$tokens->rollback();
+				break;
+			}
+
+			// There's more text on a new line, ensure spacing.
+			$text .= ' ';
```

Comment:
The lines must be separated by `\n`.

Reply:
But the intention is for a sentence to be combined if they were broken onto new lines due to width. It assumes a single line break is a continuation and double line breaks are a new entry. This is how phpDocs are parsed.

Reply:
> It assumes a single line break is a continuation and double line breaks are a new entry.

Sure, but do people writing phpDocs assume the same? This for example prevents the description from being processed as markdown (because in markdown newline is significant). I think the newlines should be kept as-is. A higher-level library may always choose to replace single newlines with space.

Reply:
https://stackoverflow.com/questions/11741302/how-do-i-put-blocks-of-php-code-into-a-phpdoc-docblock/40036952#40036952 cites various phpdoc variant sources and all suggest to support Markdown indeed in single line and multiline tags. Eg. a code example would use markdown and have significant newlines.

Drupal uses the apparantely non-standard @code tag where the content should also be significant.

Here is a Drupal example from core/core.api.php:

```
 * This will be an object of class \Drupal\Core\Config\Config, which has methods
 * for getting configuration information. For instance, if your YAML file
 * structure looks like this:
 * @code
 * enabled: '0'
 * bar:
 *   baz: 'string1'
 *   boo: 34
 * @endcode
```

Markdown examples on the link :)

Reply:
I understand the situation better, now. I'll add the `\n`.

Reply:
@JanTvrdik this was addressed